options.go

```diff
@@ -0,0 +1,289 @@
+	FcRootDrivePath    string   `long:"root-drive" description:"Path to root disk image"`
+	FcRootPartUUID     string   `long:"root-partition" description:"Root partition UUID"`
+	FcAdditionalDrives []string `long:"add-drive" description:"Path to additional drive, suffixed with :ro or :rw, can be specified multiple times"`
+	FcNicConfig        string   `long:"tap-device" description:"NIC info, specified as DEVICE/MAC"`
+	FcVsockDevices     []string `long:"vsock-device" description:"Vsock interface, specified as PATH:CID. Multiple OK"`
+	FcLogFifo          string   `long:"vmm-log-fifo" description:"FIFO for firecracker logs"`
+	FcLogLevel         string   `long:"log-level" description:"vmm log level" default:"Debug"`
+	FcMetricsFifo      string   `long:"metrics-fifo" description:"FIFO for firecracker metrics"`
+	FcDisableHt        bool     `long:"disable-hyperthreading" short:"t" description:"Disable CPU Hyperthreading"`
+	FcCPUCount         int64    `long:"ncpus" short:"c" description:"Number of CPUs" default:"1"`
+	FcCPUTemplate      string   `long:"cpu-template" description:"Firecracker CPU Template (C3 or T2)"`
+	FcMemSz            int64    `long:"memory" short:"m" description:"VM memory, in MiB" default:"512"`
+	FcMetadata         string   `long:"metadata" description:"Firecracker Metadata for MMDS (json)"`
+	FcFifoLogFile      string   `long:"firecracker-log" short:"l" description:"pipes the fifo contents to the specified file"`
+	Debug              bool     `long:"debug" short:"d" description:"Enable debug output"`
+
+	closers       []func() error
+	validMetadata interface{}
+}
+
+func (opts *options) getNetwork() ([]firecracker.NetworkInterface, error) {
+	var NICs []firecracker.NetworkInterface
+	if len(opts.FcNicConfig) > 0 {
+		tapDev, tapMacAddr, err := parseNicConfig(opts.FcNicConfig)
+		if err != nil {
+			return nil, err
+		} else {
+			allowMDDS := opts.validMetadata != nil
+			NICs = []firecracker.NetworkInterface{
+				firecracker.NetworkInterface{
+					MacAddress:  tapMacAddr,
+					HostDevName: tapDev,
+					AllowMDDS:   allowMDDS,
+				},
+			}
+		}
+	}
+	return NICs, nil
+}
+
+// constructs a list of drives from the options config
+func (opts *options) getBlockDevices() ([]models.Drive, error) {
+	blockDevices, err := parseBlockDevices(opts.FcAdditionalDrives)
+	if err != nil {
+		return nil, err
+	}
+	rootDrive := models.Drive{
+		DriveID:      firecracker.String("1"),
+		PathOnHost:   &opts.FcRootDrivePath,
+		IsRootDevice: firecracker.Bool(true),
+		IsReadOnly:   firecracker.Bool(false),
+		Partuuid:     opts.FcRootPartUUID,
+	}
+	blockDevices = append(blockDevices, rootDrive)
+	return blockDevices, nil
+}
+
+// Converts options to a usable firecracker config
+func (opts *options) getFirecrackerConfig() (firecracker.Config, error) {
+	// validate metadata json
+	if opts.FcMetadata != "" {
+		if err := json.Unmarshal([]byte(opts.FcMetadata), &opts.validMetadata); err != nil {
+			fmt.Errorf("invalid metadata: %s", err)
```

Comment:
Don't we want to return this? Looks like this does nothing other than construct an error and continue on